src/compiler/crystal/semantic/semantic_visitor.cr

```diff
@@ -419,18 +419,27 @@ abstract class Crystal::SemanticVisitor < Crystal::Visitor
   end
 
   def lookup_annotation(ann)
-    # For `@[Foo]` we actually search an annotation named `FooAnnotation`
-    path = ann.path.clone
-    path.names[-1] = "#{path.names.last}Annotation"
-
-    type = lookup_type(path)
+    # Since there's `Int::Primitive`, and now we'll have
+    # `::Primitive`, but there's no way to specify ::Primitive
+    # just yet in annotations, we temporarily hardcode
+    # that `Primitive` inside annotations means the top
+    # level primitive.
+    # We also have the same problem with File::Flags, which
```

Comment:
Could you put a `TODO:` string in here so we can grep for all todos and not miss this?

Reply:
Done!